desc/coils.py

```diff
@@ -1456,6 +1602,770 @@ def _check_type(coil0, coil):
+        else:
+            raise ValueError(
+                "cross_section_dims must be len 1 or 2, got length "
+                f"{np.size(cross_section_dims)}"
+            )
+        self._cross_section_shape = cross_section_shape
+        self._cross_section_dims = np.array(cross_section_dims).astype(float)
+        super().__init__(*args, **kwargs)
+
+    @optimizable_parameter
+    @property
+    def cross_section_dims(self):
+        """ndarray: Geometry of the coil, with cross section dimensions."""
+        return self._cross_section_dims
+
+    @cross_section_dims.setter
+    def cross_section_dims(self, new):
+        cross_section_shape = None
+        if np.size(new) == 1:
+            cross_section_shape = "circular"
+        elif np.size(new) == 2:
+            cross_section_shape = "rectangular"
+        else:
+            raise ValueError(
+                "cross_section_dims must be len 1 or 2, got length " f"{np.size(new)}"
+            )
+        self._cross_section_shape = cross_section_shape
+        self._cross_section_dims = np.array(new).astype(float)
+
+    @property
+    def cross_section_shape(self):
+        """str: Shape of the coil cross section, either 'circular' or 'rectangular'."""
+        return self._cross_section_shape
+
+    @classmethod
+    def prep_grid(cls, xsection_grid, centerline_grid=None):
+        """Prepares grids for self field computations.
+
+        Parameters
+        ----------
+        xsection_grid : LinearGrid, int or None
+            Grid used to evaluate the field on the coil cross section.
+            If an integer, uses that many equally spaced points in each
+            dimension of the cross section. If provided, must be a 2D grid with L and M
+            dimensions corresponding to the cross section spacing desired with
+            endpoint = True.
+        centerline_grid : LinearGrid or int, optional
+            Grid used to evaluate the coil centerline. If an integer, uses 2x that
+            many equally spaced points in each dimension. If provided, must be a 1D
+            grid with N dimension corresponding to the centerline spacing desired.
+
+        Returns
+        -------
+        finite_build_grid : LinearGrid
+           3D grid over coil cross section and centerline length.
+
+        """
+        # set cross section grid if not provided for u,v cross sectional coordinates
+        # L has a doubled grid size to be consistent with how M and N work
+        if xsection_grid is None:
+            L = 4
+            M = 2
+        elif isinstance(xsection_grid, numbers.Integral):
+            L = xsection_grid * 2
+            M = xsection_grid
+        elif isinstance(xsection_grid, LinearGrid):
+            if not xsection_grid.endpoint:
+                raise ValueError(
+                    "The cross section grid must have endpoint=True to compute "
+                    "the self field"
+                )
+            elif xsection_grid.N > 1:
```

Comment:
I am not sure if N=1 is what you want, don't you actually want `grid.num_zeta=1`?

an `N=1` LinearGrid has `2N+1` zeta planes in it